engine/access/rest/middleware/common_query_params.go

```diff
@@ -7,22 +7,23 @@ import (
 	"github.com/gorilla/mux"
 )
 
-const expandQueryParam = "expand"
+const ExpandQueryParam = "expand"
 const selectQueryParam = "select"
 
 // commonQueryParamMiddleware generates a Middleware function that extracts the given query parameter from the request
-// and adds it to the request context as a key value pair. The key is the query param name and the value as a string slice containing
-// all comma separated values.
-// e.g. for queryParamName "field", if the request url contains <some url>?field=value1,value2,...valueN
-// the middleware returned by commonQueryParamMiddleware will add the key - "field" to the request context with value
-// []string{"value1", "value2",..."valueN"} when it is executed
+// and adds it to the request context as a key value pair with the key as the query param name.
+// e.g. for queryParamName "fields", if the request url contains <some url>?fields=field1,field2,..fieldN,
+// the middleware returned by commonQueryParamMiddleware will add the key - "fields" to the request context with value
+// as a map containing each of the field as key and value true ["field"]->true, ["fields2"] -> true etc when it is executed
 func commonQueryParamMiddleware(queryParamName string) mux.MiddlewareFunc {
 	return func(handler http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-			if values, ok := req.URL.Query()[queryParamName]; ok {
-				values := strings.Split(values[0], ",")
-				// save the query param value in the request context
-				req = addRequestAttribute(req, queryParamName, values)
+			if value, ok := req.URL.Query()[queryParamName]; ok {
+				if len(value) > 0 && len(value[0]) > 0 {
+					values := strings.Split(value[0], ",")
+					// save the query param value in the request context
+					req = addRequestAttribute(req, queryParamName, values)
+				}
```

Comment:
What would the value of `value` be as an example? And why do we only look at the first value?

Reply:
Example for `value` is `[abcd,xyz]` for query param `/?expand=abcd%2Cxyz&select=abcd%2Cxyz`.
where `abcd,xyz` is values[0] (single element which includes the comma).